### PR TITLE
Regex improvements.

### PR DESCRIPTION
**Pull request by viktor-kurchenko**
Minor optimization related to regex patterns and a couple of typo fixes.

### CODE REVIEW

File: pkg/policy/api/cidr_test.go [hunk 1]
```diff
@@ -83,11 +83,9 @@ func (s *PolicyAPITestSuite) TestGetAsEndpointSelectors(c *C) {
...
+var cidrRegex = regexp.MustCompile(`^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)/([0-9]|[1-2][0-9]|3[0-2])$|^s*((([0-9A-Fa-f]{1,4}:){7}(:|([0-9A-Fa-f]{1,4})))|(([0-9A-Fa-f]{1,4}:){6}:([0-9A-Fa-f]{1,4})?)|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4})?):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){0,2}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){0,3}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){0,4}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:)(((:[0-9A-Fa-f]{1,4}){0,5}):([0-9A-Fa-f]{1,4})?))|(:(:|((:[0-9A-Fa-f]{1,4}){1,7}))))(%.+)?s*/([0-9]|[1-9][0-9]|1[0-1][0-9]|12[0-8])$`)
```
Comment (nathanjsweet) [inline pkg/policy/api/cidr_test.go:86]: I just diff'ed these two strings, and it appears you made changes, can you explain why?

**Review by nathanjsweet — CHANGES_REQUESTED** [review]

File: pkg/policy/api/cidr_test.go (hunk 1 above)
Comment (viktor-kurchenko) [reply pkg/policy/api/cidr_test.go:86]: Yeah, I dropped redundant characters:
1. `\/` -> `/`
2. `{0,1}` -> `?`
3. `{1}` -> ``

Actually, it was my IDE regex analyzer)
But if you think it's risky I'm fine to revert it.

**Review by bimmlerd — APPROVED** [review]
LGTM

**Review by dylandreimerink — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Review by bimmlerd — CHANGES_REQUESTED** [review]
I think we don't do merges from main into the PR - Instead rebase the PR on current main and force push it :)

**Comment by viktor-kurchenko** [comment]
> I think we don't do merges from main into the PR - Instead rebase the PR on current main and force push it :)

Sorry, didn't know that...

**Comment by bimmlerd** [comment]
> Sorry, didn't know that...

That's fine - you still nned do it though; running the tests doesn't really help at the moment, since the rebasing will require you to run them again.
Specifically, off the top of my head I think you need to
```shell-session
git checkout regex_improved # checkout the branch if you haven't
git fetch origin            # or whatever your cilium/cilium remote is called, to ensure you're up to date
git reset 'HEAD^' --hard    # to drop the merge commit
git rebase origin/main      # to rebase your commit on main
git push --force-with-lease your-fork HEAD # push to your fork
```
alternatively, you can also just create a new branch off main, and cherry-pick the relevant commit onto it, but then you need to fiddle around a bit to be able to push it to this branch.

**Comment by bimmlerd** [comment]
> > push --force-with-lease
> 
> David thanks! I did it.

:sweat_smile: It should just be one commit in this PR if you rebased correctly

**Review by bimmlerd — APPROVED** [review]
commits lgtm now.

**Pull request closed without merging**